tests/contrib/django/test_django_snapshots.py

```diff
@@ -98,16 +98,20 @@ def test_middleware_trace_partial_based_view(client):
 
 
 @pytest.mark.django_db
-@snapshot(
-    variants={
-        "18x": django.VERSION < (1, 9),
-        "111x": (1, 9) <= django.VERSION < (1, 12),
-        "21x": (1, 12) < django.VERSION < (2, 2),
-        "": django.VERSION >= (2, 2),
-    }
-)
-def test_safe_string_encoding(client):
-    assert client.get("/safe-template/").status_code == 200
+def test_safe_string_encoding(client, snapshot_context):
+    """test_safe_string_encoding.
+    If we use @snapshot decorator in a Django snapshot test, the first test adds DB creation traces
+    """
+    client.get("/safe-template/")
```

Comment:
```suggestion
```

?

Reply:
Do we need this initial call?

Reply:
yes, we need it :) 

Without this line:
```
At request <Request GET /test/session/snapshot >:
   At snapshot (token='tests.contrib.django.test_django_snapshots.test_safe_string_encoding'):
    - Directory: /snapshots
    - CI mode: 0
    - Trace File: /snapshots/tests.contrib.django.test_django_snapshots.test_safe_string_encoding.json
    - Stats File: /snapshots/tests.contrib.django.test_django_snapshots.test_safe_string_encoding_tracestats.json
    At compare of 1 expected trace(s) to 1 received trace(s):
     At trace 'django.request' (33 spans):
Received more spans (35) than expected (33). Received unmatched spans: 'sqlite.query', 'django.cache'
```

Reply:
can we add a comment here on why we have this extra call?

it is also likely going to be an issue if someone tries to do `pytest ... -k test_psycopg_query_default`, the initial database setup/migration will run and that test will fail (e.g. this fix/extra call will only work if this test is called first).

I am ok with it for now if we can add a comment that that is why we are doing that and we can come back later to try and solve the issue.

Reply:
`pytest ... -k test_psycopg_query_default` doesn't need a extra call such as test_safe_string_encoding. With these changes ~~I think~~  I hope  both tests are idempotent.

Docstring updated :+1: